### PR TITLE
[Modules] Updated Disk Encryption Set to latest published API version (2022-07-02)

### LINKED ISSUE #2632: [Failed pipeline] Compute: DiskEncryptionSets
failed run: https://github.com/Azure/ResourceModules/actions/runs/4048059986

### PR DESCRIPTION
**Pull request by ahmadabdalla**
Closes #2632 

# Description

Updated the DES module to the latest API version to fix failing pipeline: 

Added support for:
- federatedClientId: Multi-tenant application client id to access key vault in a different tenant. 
- User Assigned Managed Identity support. Set the system assigned to default to true.

| Pipeline |
| - |
|[Compute: DiskEncryptionSets](https://github.com/Azure/ResourceModules/actions/workflows/ms.compute.diskencryptionsets.yml) |

# Type of Change

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

# Checklist

- [ ] I'm sure there are no other open Pull Requests for the same update/change
- [ ] My corresponding pipelines / checks run clean and green without any errors or warnings
- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (readme)
- [ ] I did format my code

### CODE REVIEW

File: modules/Microsoft.Compute/diskEncryptionSets/.test/common/dependencies.bicep [hunk 1]
```diff
@@ -16,7 +16,7 @@ resource keyVault 'Microsoft.KeyVault/vaults@2022-07-01' = {
             name: 'standard'
         }
         tenantId: tenant().tenantId
-        enablePurgeProtection: null
+        enablePurgeProtection: true
```
Comment (AlexanderSehr) [inline modules/Microsoft.Compute/diskEncryptionSets/.test/common/dependencies.bicep:19]: Does it have to be true? If not I strongly recommend to set it to `false`. Not that we don't generate a new name each time we run. As you ran a test case there's now already a fair chance we cannot re-run the test for the next couple days :D

Comment (ahmadabdalla) [reply modules/Microsoft.Compute/diskEncryptionSets/.test/common/dependencies.bicep:19]: Hmmm.. it was complaining it needed to be true unfortunately.. given it was working before, maybe the new API mandates this.. but I was blocked by the purge protection on KV must be set to true

Comment (AlexanderSehr) [reply modules/Microsoft.Compute/diskEncryptionSets/.test/common/dependencies.bicep:19]: That's alright (might be new) - but then we must adopt the KeyVault deployment approach we use for Purge Protected Key Vaults like in the Batch Module's Encryption test

Comment (ahmadabdalla) [reply modules/Microsoft.Compute/diskEncryptionSets/.test/common/dependencies.bicep]: No problem. Aligned it to the batch module for baseTime usage.

**Review by AlexanderSehr — APPROVED** [review]

**Pull request merged**